### PR TITLE
Added format support to Text

### PR DESCRIPTION
**Pull request by sebastianbaginski**
Will be needed in `DateTime`.
I tried to move `Format.ooc` out of SDK but it's not that easy...
@marcusnaslund

### CODE REVIEW

File: source/base/Text.ooc [hunk 1]
```diff
@@ -182,6 +182,16 @@ Text: cover {
 		printf("%s\n", this _buffer _backend pointer as Char*)
 		this free(Owner Receiver)
 	}
+	format: func (args: ...) -> This {
+		t := this take()
+		string := t toString()
```
Comment (sebastianbaginski) [inline source/base/Text.ooc]: It depends on `String` for now, but at least `DateTime` and `TimeSpan` can be implemented without `String` now.

**Comment by marcusnaslund** [comment]
> I tried to move Format.ooc out of SDK but it's not that easy...

Actually, @simonmika had some idea about moving `Text` into the sdk instead. Nothing's decided there yet though. We should talk about that.

File: source/base/Text.ooc (hunk 1 above)
Comment (marcusnaslund) [inline source/base/Text.ooc]: Does that mean we will be able to eliminate those leaks?

**Comment by marcusnaslund** [comment]
@tomasmahlberg can you please check the `OwnedBuffer` usage here?

File: source/base/Text.ooc [hunk 2]
```diff
@@ -182,6 +182,16 @@ Text: cover {
 		printf("%s\n", this _buffer _backend pointer as Char*)
 		this free(Owner Receiver)
 	}
+	format: func (args: ...) -> This {
+		t := this take()
```
Comment (tomasmahlberg) [inline source/base/Text.ooc]: This is maybe a bit unnecessary since you only use `t` in one place, better just use `this take()` explicitly perhaps?

Comment (sebastianbaginski) [inline source/base/Text.ooc]: OK

**Comment by sebastianbaginski** [comment]
Alright, but since you had that issue about removing SDK I thought I can help you a bit ;)

File: source/base/Text.ooc [hunk 3]
```diff
@@ -182,6 +182,16 @@ Text: cover {
...
+		string := t toString()
+		resultString := string format(args)
+		string free()
+		result := This new(resultString) copy()
+		resultString free()
+		this free(Owner Receiver)
+		result give()
```
Comment (tomasmahlberg) [inline source/base/Text.ooc]: This `give()` could be moved to the end of the line where you assign `result`, but it's not important.

File: source/base/Text.ooc (hunk 1 above)
Comment (sebastianbaginski) [inline source/base/Text.ooc]: That's my goal.

**Comment by marcusnaslund** [comment]
Right, there's still a lot of stuff that should be moved out, but maybe some stuff that should be left. Let's discuss this soon.

File: source/base/Text.ooc (hunk 3 above)
Comment (sebastianbaginski) [inline source/base/Text.ooc]: So the `give` is needed, it is not enough to call `copy` ?

Comment (marcusnaslund) [inline source/base/Text.ooc]: Let's do it, it saves us a line.

File: source/base/Text.ooc (hunk 2 above)
Comment (sebastianbaginski) [inline source/base/Text.ooc]: @tomasmahlberg can you check again ?

File: source/base/Text.ooc (hunk 3 above)
Comment (tomasmahlberg) [inline source/base/Text.ooc]: Yes, I have no idea actually, you'd have to trace this function to figure it out...

Comment (sebastianbaginski) [inline source/base/Text.ooc]: `copy` already sets owner to receiver, so I think `give` is not needed here.

**Comment by sebastianbaginski** [comment]
Updated.

**Pull request merged**